feat: delete account

afterburn opened the pull request:
Data layer specific code for the delete account operation

tipogi commented on src/core/services/local/user/user.ts:
```diff
@@ -0,0 +1,59 @@
+            // Stream tables
+            Core.PostStreamModel.table.clear(),
+            Core.UserStreamModel.table.clear(),
+            Core.TagStreamModel.table.clear(),
```
maybe create in the abstract class that function of `clear()`and like this we do not reference `.table`?

afterburn commented:
@tipogi resolved your feedback:
- Moved delete account to profile system instead of user system.
- Added `clear()` to base models so we no longer have to call `.table.clear()`.
- Temporarily passing Infinity to `homeserver.list` so we make to delete all files instead of only first `500` until `homeserver.list` supports pagination/cursor.
